test_syncobj.py

```diff
@@ -357,6 +372,92 @@ def test_syncThreeObjectsLeaderFail():
 	o2._destroy()
 	o3._destroy()
 
+def sync_noflush_novote(journalFile2Enabled):
+	# Test that if flushing is disabled, the node won't vote until the maximum timeout has been exceeded
+	# o1's timeout is set to 0.5 seconds, so it will call for an election almost immediately.
+	# o2's timeout is set to 2.4 seconds, so it will not call for an election before o1, and it will ignore o1's request_vote messages.
+	# Specifically, o2 is expected to ignore the messages until 1.1 * timeout, i.e. including the one sent by o1 after 2.5 seconds, except for updating its term.
+	# Note that o1 has flushing enabled but o2 doesn't!
+
+	def tick(o1, o2, startTime, totalTickTime, sleepTime):
```

Comment:
time-specific tests will randomly fails sometimes - you should mock time or not rely on time at all

Reply:
Oh, great idea. I'll implement that. :+1: